processor/src/test/java/com/linecorp/decaton/processor/runtime/ProcessorSubscriptionTest.java

```diff
@@ -106,4 +136,148 @@ public void testStateTransition() throws Exception {
                                    State.SHUTTING_DOWN,
                                    State.TERMINATED), states);
     }
+
+    private ProcessorSubscription subscriptionForCommitTest(BlockingQueue<Runnable> tasks) {
+        SubscriptionScope scope = scope("topic");
+        ProcessorSubscription subscription = new ProcessorSubscription(
+                scope,
+                () -> consumerMock,
+                null,
+                scope.props(),
+                null,
+                contextsMock) {
+            @Override
+            public void run() {
```

Comment:
I understand this, but overriding `run` for test (though `run` contains important logic actually) looks like sign that we should refactor ProcessorSubscription to separate its responsibility into few components, for better testability...

Reply:
Agree. Do you want me to try that refactoring in this PR or leave it for later effort?

Reply:
Hm, I think it will make big diff, and we shouldn't include 2 significant changes about ProcessorSubscription in 1 PR

So let's do later